src/common/ossl_helpers.c

```diff
@@ -46,30 +46,58 @@ static void fetch_ossl_objects(void) {
 }
 
 static void free_ossl_objects(void) {
-	OSSL_FUNC(EVP_MD_free)(sha256_ptr);
-	sha256_ptr = NULL;
-	OSSL_FUNC(EVP_MD_free)(sha384_ptr);
-	sha384_ptr = NULL;
-	OSSL_FUNC(EVP_MD_free)(sha512_ptr);
-	sha512_ptr = NULL;
-	OSSL_FUNC(EVP_MD_free)(sha3_256_ptr);
-	sha3_256_ptr = NULL;
-	OSSL_FUNC(EVP_MD_free)(sha3_384_ptr);
-	sha3_384_ptr = NULL;
-	OSSL_FUNC(EVP_MD_free)(sha3_512_ptr);
-	sha3_512_ptr = NULL;
-	OSSL_FUNC(EVP_MD_free)(shake128_ptr);
-	shake128_ptr = NULL;
-	OSSL_FUNC(EVP_MD_free)(shake256_ptr);
-	shake256_ptr = NULL;
-	OSSL_FUNC(EVP_CIPHER_free)(aes128_ecb_ptr);
-	aes128_ecb_ptr = NULL;
-	OSSL_FUNC(EVP_CIPHER_free)(aes128_ctr_ptr);
-	aes128_ctr_ptr = NULL;
-	OSSL_FUNC(EVP_CIPHER_free)(aes256_ecb_ptr);
-	aes256_ecb_ptr = NULL;
-	OSSL_FUNC(EVP_CIPHER_free)(aes256_ctr_ptr);
-	aes256_ctr_ptr = NULL;
+	/* Always check argument is non-NULL before calling EVP_*_free
+	 * to avoid OpenSSL functions being used when they are
+	 * overridden with OQS_*_set_callbacks.
+	 */
+	if (sha256_ptr) {
```

Comment:
Wouldn't it be better to transform this into a macro (or add an IF_NONNULL variant of "OSSL_FUNC") to make this more easily readable?

Reply:
Instead of a macro, I added inline functions that work in a similar fashion as GCC's cleanup functions.

Reply:
Looks cleaner and easier to maintain this way. Thanks!